crates/interfaces/src/p2p/bodies/client.rs

```diff
@@ -2,28 +2,27 @@ use std::pin::Pin;
 
 use crate::p2p::{download::DownloadClient, error::PeerRequestResult, priority::Priority};
 use futures::Future;
+use reth_eth_wire::BlockBody;
 use reth_primitives::H256;
 use tokio::sync::oneshot::error::RecvError;
 
 /// The bodies future type
-pub type BodiesFuture<T> =
-    Pin<Box<dyn Future<Output = Result<PeerRequestResult<T>, RecvError>> + Send + Sync>>;
+pub type BodiesFut = Pin<
+    Box<dyn Future<Output = Result<PeerRequestResult<Vec<BlockBody>>, RecvError>> + Send + Sync>,
+>;
 
 /// A client capable of downloading block bodies.
 #[auto_impl::auto_impl(&, Arc, Box)]
 pub trait BodiesClient: DownloadClient {
     /// The bodies type
-    type Output;
+    type Output: Future<Output = Result<PeerRequestResult<Vec<BlockBody>>, RecvError>> + Sync + Send;
```

Comment:
this should only be `Result<PeerRequestResult<Vec<BlockBody>>` since RequestError can be created from RecvErr

Reply:
with the conversion it's likely that we need to roll some helper types and implement future manually